### PR TITLE
fix(cli): Test workflow retry with and without persistence. Fixes #2850

### LINKED ISSUE #2850: Fix flakey TestWorkflowRetryNoPersistence
# Summary

This test is flakey because it is using banned image `whalesay` rather than `cowsay:v1`

# Motivation

This should have failed validation in fact. Not sure why it did not.

Can we also find out why this test is a special case? I can't see why this test has a conditional on persistence - it should work fine without it.

# Proposal

* Change image to cowsay.
* Determine why this test case is a special - and hopefully make it non-special.

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by mark9white**
Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [X] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [X] I've signed the CLA.
* [X] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [X] My organization is added to USERS.md.

### CODE REVIEW

File: test/e2e/cli_test.go [hunk 1]
```diff
@@ -466,7 +466,10 @@ func (s *CLISuite) TestWorkflowRetryNoPersistence() {
 		// When this is the case, this behavior is tested in cli_with_server_test.go
 		s.T().SkipNow()
 	}
+	WorkflowRetryCommon(s.E2ESuite)
+}
 
+func WorkflowRetryCommon(s fixtures.E2ESuite) {
```
Comment (simster7) [inline test/e2e/cli_test.go:472]: Should we be putting all these in a `cli_common.go` file?

Comment (alexec) [reply test/e2e/cli_test.go:472]: I don't like this. E2ESuite is an abstract base class and we shouldn't be passing it around like a normal variable.

Comment (mark9white) [reply test/e2e/cli_test.go:478]: I think the cli_common.go is a good idea. @alexec do you have an alternate idea how to structure these common tests?

**Comment by alexec** [comment]
This PR looks stale?

**Comment by mark9white** [comment]
Yes, I think made moot by #2941 - I will close it.

**Pull request closed without merging**